contracts/connection-router/src/state.rs

```diff
@@ -275,6 +289,12 @@ impl KeyDeserialize for &ChainName {
     }
 }
 
+impl ChainName {
+    pub fn as_str(&self) -> &str {
+        self.0.as_str()
+    }
+}
```

Comment:
Can we just implement `AsRef` on ChainName to get a reference to the underlying string, and then just call as_str() on the reference?

Reply:
implemented `AsRef<str>` and used it directly